### PR TITLE
[ENG-5139] Added navbar tests to collections and projects details pages

### PR DESCRIPTION
**Pull request by Ramyapatil712**
## Purpose
Update our selenium test to add new navbar tests to collections discover page and project details page

Product team requested to add these tests to verify search, donate and support link on project details page and collections discover page.

## Summary of Changes

For Collections-

- Updated collections.py - added few locators for Search, Add to collections and Donate buttons
- test_navbar.py — Added new classes TestCollectionsNavbarLoggedOut and TestCollectionsNavbarLoggedIn

For Project Details page

Updated test_navbar.py- Added new tests
navbars.py- Added locator for my_projects link 
 

## Reviewer's Actions
`git fetch <remote> pull/262/head:feature/add-navbartests-to-collections-project`

Run this test using
`pytest tests/test_navbar.py::TestCollectionsNavbarLoggedIn -sv `
`pytest tests/test_navbar.py::TestCollectionsNavbarLoggedOut -sv `
`pytest tests/test_navbar.py::TestProjectsNavbarLoggedIn -sv `

## Testing Changes Moving Forward
N/A

## Ticket

https://openscience.atlassian.net/browse/ENG-5139

### CODE REVIEW

File: tests/test_navbar.py [hunk 1]
```diff
@@ -316,3 +318,93 @@ def assert_donate_page(driver, donate_page):
...
+
+@markers.smoke_test
+@markers.core_functionality
+class TestCollectionsNavbarLoggedOut():
+    @pytest.fixture
+    def provider(self, driver):
+        return osf_api.get_provider(type='collections', provider_id='selenium')
+
+    @pytest.fixture()
+    def collectionsdiscover_page(self, driver, provider):
```
Comment (felliott) [inline tests/test_navbar.py]: Nitpick: can you add an underscore between `collections` and `discover`

File: tests/test_navbar.py [hunk 2]
```diff
@@ -316,3 +318,93 @@ def assert_donate_page(driver, donate_page):
...
+        discover_page = CollectionDiscoverPage(driver, provider=provider)
+        discover_page.goto()
+        discover_page.loading_indicator.here_then_gone()
+        return discover_page
+
+    def test_search_link(self, driver, collectionsdiscover_page):
```
Comment (felliott) [inline tests/test_navbar.py]: Ditto.

File: tests/test_navbar.py [hunk 3]
```diff
@@ -316,3 +318,93 @@ def assert_donate_page(driver, donate_page):
...
+        collectionsdiscover_page.search_link.click()
```
Comment (felliott) [inline tests/test_navbar.py]: Ditto.

File: tests/test_navbar.py [hunk 4]
```diff
@@ -316,3 +318,93 @@ def assert_donate_page(driver, donate_page):
...
+         default_project_page.goto()
+         return default_project_page
+
+    @pytest.fixture()
+    def page(self, driver, project_with_file):
+        page = ProjectPage(driver, guid=project_with_file.id)
+        page.goto()
+        return page
+
+    def test_my_projects_link(self, page, driver, fake):
```
Comment (felliott) [inline tests/test_navbar.py]: What is `fake` in the arg list?

**Review by felliott — CHANGES_REQUESTED** [review]
Some nitpicks and a few questions. There are a few small style issues i didn't mention that will be fixed up by running `isort` and `black`.

File: tests/test_navbar.py (hunk 1 above)
Comment (Ramyapatil712) [reply tests/test_navbar.py]: sure. will do

File: tests/test_navbar.py (hunk 4 above)
Comment (Ramyapatil712) [reply tests/test_navbar.py]: I copied this piece of code from another test. Will check and come back on this

**Comment by Ramyapatil712** [comment]
@felliott , @jh27539 and @cslzchen Done all the changes suggested and used black for formatting. Please review the changes

File: .python-version [hunk 5]
```diff
@@ -0,0 +1 @@
+Sel-Env-3.9.7
```
Comment (felliott) [inline .python-version]: This shouldn't be checked in.  Maybe add `.python-version` to `.gitignore` so others don't encounter it.

File: tests/test_navbar.py (hunk 4 above)
Comment (Ramyapatil712) [reply tests/test_navbar.py]: Argument fake is not required here as we are not generating any fake names or ids. So I have removed it from the code.

**Pull request merged**